distributed/scheduler.py

```diff
@@ -4986,13 +4987,14 @@ async def remove_worker(self, comm=None, address=None, safe=False, close=True):
 
             ws: WorkerState = parent._workers_dv[address]
 
-            self.log_event(
-                ["all", address],
-                {
-                    "action": "remove-worker",
-                    "processing-tasks": dict(ws._processing),
-                },
-            )
+            event_msg = {
+                "action": "remove-worker",
+                "processing-tasks": dict(ws._processing),
+            }
+            self.log_event(address, event_msg.copy())
+            event_msg["worker"] = address
+            self.log_event("all", event_msg)
```

Comment:
Not having the worker in the all events makes the entire message useless. I wouldn't mind having the redundancy of the key in the worker specific topic but we made a difference for add-worker as well so I aligned.